### PR TITLE
feature/#5597/[Events]Add possibility to choose filters

### PR DESCRIPTION
**Pull request by SokhaTaras**

### CODE REVIEW

**Review by KKatiaa — APPROVED** [review]

**Review by NadiiaStoiko — APPROVED** [review]

**Review by atsa21 — APPROVED** [review]

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 1]
```diff
@@ -89,22 +95,58 @@ export class EventsListComponent implements OnInit, OnDestroy {
     });
   }
 
+  updateSelectedFilters(value: any, event): void {
+    const existingFilterIndex = this.selectedFilters.indexOf(value);
+    if (event.isUserInput && !event.source.selected && existingFilterIndex !== -1) {
+      this.selectedFilters.splice(existingFilterIndex, 1);
+    } else if (event.isUserInput && event.source.selected && existingFilterIndex === -1) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: don't use 'else if' here

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 2]
```diff
@@ -89,22 +95,58 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
 
-  public isSelectedOptions(optionsList: any): void {
+  public isSelectedOptions(optionsList: any, dropdownName: string): void {
     this.optionsList = optionsList;
-    this.allSelected
-      ? this.optionsList.options.forEach((item: MatOption) => item.select())
-      : this.optionsList.options.forEach((item: MatOption) => item.deselect());
+    if (this.allSelectedFlags[dropdownName]) {
+      this.optionsList.options.forEach((item: MatOption) => {
+        if (item.value !== 0) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts:138]: !item.value

File: src/app/main/component/events/components/events-list/events-list.component.ts (hunk 1 above)
Comment (SokhaTaras) [reply src/app/main/component/events/components/events-list/events-list.component.ts]: fixed

File: src/app/main/component/events/components/events-list/events-list.component.ts (hunk 2 above)
Comment (SokhaTaras) [reply src/app/main/component/events/components/events-list/events-list.component.ts:138]: fixed

**Review by katia9261 — APPROVED** [review]

**Review by ruslan-teh — APPROVED** [review]

**Pull request merged**